contracts/base/strategies/iron-fold/IronFoldStrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: ISC
+/**
+* By using this software, you understand, acknowledge and accept that Tetu
+* and/or the underlying software are provided “as is” and “as available”
+* basis and without warranties or representations of any kind either expressed
+* or implied. Any use of this open source software released under the ISC
+* Internet Systems Consortium license is done at your own risk to the fullest
+* extent permissible pursuant to applicable law any and all liability as well
+* as all warranties, including any fitness for a particular purpose with respect
+* to Tetu and/or the underlying software and the use thereof are disclaimed.
+*/
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../StrategyBase.sol";
+import "../../../third_party/iron/IIronChef.sol";
+import "../../../third_party/iron/IRMatic.sol";
+import "../../../third_party/iron/CompleteRToken.sol";
+
+
+/// @title Abstract contract for Iron strategy implementation
+/// @author belbix
+abstract contract IronFoldStrategyBase is StrategyBase {
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  // ************ VARIABLES **********************
+  /// @notice Strategy type for statistical purposes
+  string public constant STRATEGY_NAME = "IronFoldStrategyBase";
+  /// @notice Version of the contract
+  /// @dev Should be incremented when contract changed
+  string public constant VERSION = "1.0.0";
+  /// @dev Placeholder, for non full buyback need to implement liquidation
+  uint256 private constant _BUY_BACK_RATIO = 10000;
+
+  /// @notice RToken address
+  address public rToken;
+  /// @notice Iron Controller address
+  address public ironController;
+
+  uint256 public borrowTargetFactorNumerator;
```

Comment:
please describe in comments these variables purposes

Reply:
Done